### PR TITLE
aptos: initial commit

### PR DESCRIPTION
**Pull request by evan-gray**
see aptos/README.md for checklist status. for MVP, this is missing events.

### CODE REVIEW

File: aptos/router/sources/integrator.move [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    public entry fun transfer_admin(admin: &signer, integrator_addr: address, new_admin: address) acquires AdminConfig {
+        // MUST check that the caller is the current admin and there is not a pending transfer.
+        let admin_config = &mut AdminConfig[integrator_addr];
+        assert!(admin_config.admin_addr.contains(&signer::address_of(admin)));
+        // If possible, MUST NOT allow the admin to discard admin via this command.
+        assert!(new_admin != @0x0);
+        // Initiates the first step of a two-step process in which the current admin (to cancel) or new admin must claim.
+        // .fill will ensure there was not a pending transfer, as it muust be empty in order to fill
```
Comment (bingyuyap) [inline aptos/router/sources/integrator.move:106]: nit 
```suggestion
        // .fill will ensure there was not a pending transfer, as it must be empty in order to fill
```

File: aptos/router/sources/integrator.move [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        admin_config.pending_admin_addr.fill(new_admin);
+    }
+    
+    public entry fun claim_admin(new_admin: &signer, integrator_addr: address) acquires AdminConfig {
+        // MUST check that the caller is the current admin OR the pending admin.
+        let admin_config = &mut AdminConfig[integrator_addr];
+        let new_admin_addr = signer::address_of(new_admin);
+        assert!(admin_config.admin_addr.contains(&new_admin_addr) || admin_config.pending_admin_addr.contains(&new_admin_addr));
```
Comment (bingyuyap) [inline aptos/router/sources/integrator.move:114]: What error does it throw when this fails?

File: aptos/router/sources/router.move [hunk 3]
```diff
@@ -0,0 +1,891 @@
+module router::router {
+    use aptos_framework::aptos_hash;
+    use router::integrator;
+    use router::universal_address::{Self, UniversalAddress};
+    use std::bcs;
+    use std::signer;
+    use std::table::{Self, Table};
+    use std::vector;
+
+    const MAX_U128: u128 = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
```
Comment (bingyuyap) [inline aptos/router/sources/router.move:10]: maybe we can move this out to a util/const file

**Review by bingyuyap — DISMISSED** [review]
just a nit and maybe for asserts we want to define the errors

**Review by bingyuyap — APPROVED** [review]
thanks for adding the errors!

**Comment by evan-gray** [comment]
will come back for events

**Pull request merged**